### ISSUE TITLE
2.7.1 - Package missing

### ISSUE THREAD

**Issue by indieboyalex** [muxinc/mux-stats-sdk-exoplayer#206]
Exactly the same as https://github.com/muxinc/mux-stats-sdk-exoplayer/issues/202 but with 2.7.1
<img width="551" alt="Screenshot 2022-04-29 at 09 13 02" src="https://user-images.githubusercontent.com/851270/165900139-7a371b69-a6b5-4505-a104-2f168e07edfa.png">

**Comment by daytime-em** [comment]
We're very sorry you're still having issues with this. We are unable to replicate this on our side. Just to be clear, are you still experiencing build issues as well? We don't expect the classes to be in this `aar` anymore, but they come in via a transitive dependency.

Can you verify whether or not  `'com.mux:stats.muxcore:7.0.11'` is anywhere in your classpath? For example the dependencies for our Demo App have it here: 
<img width="472" alt="Screen Shot 2022-04-29 at 11 00 28 AM" src="https://user-images.githubusercontent.com/319221/165998308-3c2ab7c8-e380-4cc1-945a-b150e75c78a2.png">

If `stats.muxcore` is not showing up for you, would you be willing to share your build configuration?

**Comment by bolot** [comment]
I experienced the same issue, but in my case, this turned out to be due to a misconfiguration of the Mux maven repository.

In my app `build.gradle`, I have the equivalent of:

```gradle
api "com.mux.stats.sdk.muxstats:MuxExoPlayer_r2_17_1:2.7.1"
```

In my root `build.gradle`, I had this:

```gradle
allprojects {
    repositories {
        // ... other repos
        maven {
            url 'https://muxinc.jfrog.io/artifactory/default-maven-release-local'
            content {
                includeGroup 'com.mux.stats.sdk.muxstats'
            }
        }
    }
}
```

Note that the offending library has a different group name: `com.mux`. Adding it fixed the issue for me:

```gradle
allprojects {
    repositories {
        // ... other repos
        maven {
            url 'https://muxinc.jfrog.io/artifactory/default-maven-release-local'
            content {
                includeGroup 'com.mux.stats.sdk.muxstats'
                includeGroup 'com.mux' // <-- this fixed the issue
            }
        }
    }
}
```

**Comment by indieboyalex** [comment]
I managed to resolve this by removing a rogue '@aar' from the gradle declaration.

HOWEVER, I now have a bigger issue that prevents us from building our app.

I see from the dependencies that you look to be importing the exoplayer libraries.  Is it possible to prevent this, as we are using a slightly custom build of the exoplayer libraries and your imports are preventing our project from building due to duplicate class declarations.

<img width="706" alt="Screenshot 2022-05-03 at 09 36 37" src="https://user-images.githubusercontent.com/851270/166418551-b86327a8-cd3d-4bd7-a756-d7856418b7c0.png">

<img width="1555" alt="Screenshot 2022-05-03 at 09 39 51" src="https://user-images.githubusercontent.com/851270/166418899-5514f1d7-adaa-4b75-b2b2-247d58536677.png">

**Comment by indieboyalex** [comment]
For the time being I'm excluding the transitive dependencies manually:

`
api (Libs.mux) {
        exclude group: "com.google.android.exoplayer"
        exclude group: "org.json"
    }
`

**Comment by indieboyalex** [comment]
Additionally, once we have excluded the transitive dependencies, we crash with:

`java.lang.NoClassDefFoundError: Failed resolution of: Lcom/google/android/exoplayer2/source/hls/HlsManifest;
        at com.mux.stats.sdk.muxstats.MuxStatsExoPlayer.onTimelineChanged(Unknown Source:14)
        at com.mux.stats.sdk.muxstats.MuxStatsExoPlayer.onTimelineChanged(Unknown Source:2)`
        
This no doubt being because our app does not use HLS and therefore we do not include the useless HLS part of exoplayer (as the exoplayer team intend us to be able to do).

**Comment by daytime-em** [comment]
Thank you both for your detailed feedback!

@bolot For the time being, please continue using both content filters for our maven repository. We may update our `groupId`s for consistency in the future. Sorry for any inconvenience this has caused.

@iNdieboyjeff We have fixes for the crash and the duplicate class errors. The next update will be able to tolerate the absence of the HLS module. I'll close this issue once the update is out.

Thank you both again!

**Comment by daytime-em** [comment]
@iNdieboyjeff your issues with crashes and duplicate class errors should be resolved in v2.7.2 of the SDK. The Data SDK should no longer declare any exoplayer depedencies for runtime. I think you should probably keep filtering out the `exoplayer` package like you're doing, to avoid unwanted symbols in your compile classpath.

Thanks again for your feedback!

**Comment by indieboyalex** [comment]
@daytime-em I've updated to 2.8.0 and still crashing due to dependency:

`2022-06-10 10:40:22.496 11512-11512/com.upst.hayu E/AndroidRuntime: FATAL EXCEPTION: main
    Process: XXXXXXXXXXX, PID: 11512
    java.lang.NoClassDefFoundError: Failed resolution of: Lcom/google/android/exoplayer2/source/hls/HlsManifest;
        at com.mux.stats.sdk.muxstats.MuxBaseExoPlayer.isHlsExtensionAvailable(Unknown Source:4)
        at com.mux.stats.sdk.muxstats.MuxBaseExoPlayer.handleHlsManifest(Unknown Source:2)
        at com.mux.stats.sdk.muxstats.MuxStatsExoPlayer.onTimelineChanged(Unknown Source:8)
        at com.mux.stats.sdk.muxstats.MuxStatsExoPlayer.onTimelineChanged(Unknown Source:2)
        at com.google.android.exoplayer2.analytics.AnalyticsCollector.lambda$onTimelineChanged$33(AnalyticsCollector.java:576)
        at com.google.android.exoplayer2.analytics.AnalyticsCollector$$ExternalSyntheticLambda4.invoke(Unknown Source:6)
        at com.google.android.exoplayer2.util.ListenerSet$ListenerHolder.invoke(ListenerSet.java:261)
        at com.google.android.exoplayer2.util.ListenerSet.lambda$queueEvent$0(ListenerSet.java:170)
        at com.google.android.exoplayer2.util.ListenerSet$$ExternalSyntheticLambda1.run(Unknown Source:6)
        at com.google.android.exoplayer2.util.ListenerSet.flushEvents(ListenerSet.java:191)
        at com.google.android.exoplayer2.util.ListenerSet.sendEvent(ListenerSet.java:206)
        at com.google.android.exoplayer2.analytics.AnalyticsCollector.sendEvent(AnalyticsCollector.java:897)
        at com.google.android.exoplayer2.analytics.AnalyticsCollector.onTimelineChanged(AnalyticsCollector.java:573)
        at com.google.android.exoplayer2.ExoPlayerImpl.lambda$updatePlaybackInfo$9(ExoPlayerImpl.java:1268)
        at com.google.android.exoplayer2.ExoPlayerImpl$$ExternalSyntheticLambda9.invoke(Unknown Source:6)
        at com.google.android.exoplayer2.util.ListenerSet$ListenerHolder.invoke(ListenerSet.java:261)
        at com.google.android.exoplayer2.util.ListenerSet.lambda$queueEvent$0(ListenerSet.java:170)
        at com.google.android.exoplayer2.util.ListenerSet$$ExternalSyntheticLambda1.run(Unknown Source:6)
        at com.google.android.exoplayer2.util.ListenerSet.flushEvents(ListenerSet.java:191)
        at com.google.android.exoplayer2.ExoPlayerImpl.updatePlaybackInfo(ExoPlayerImpl.java:1366)
        at com.google.android.exoplayer2.ExoPlayerImpl.setMediaSourcesInternal(ExoPlayerImpl.java:1577)
        at com.google.android.exoplayer2.ExoPlayerImpl.setMediaSources(ExoPlayerImpl.java:433)
        at com.google.android.exoplayer2.ExoPlayerImpl.setMediaSources(ExoPlayerImpl.java:429)
        at com.google.android.exoplayer2.ExoPlayerImpl.setMediaSource(ExoPlayerImpl.java:416)
        at com.google.android.exoplayer2.SimpleExoPlayer.setMediaSource(SimpleExoPlayer.java:1119)`
